spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/SimpleRouteLocator.java

```diff
@@ -138,7 +138,7 @@ protected Route getRoute(ZuulRoute route, String path) {
 		}
 		String targetPath = path;
 		String prefix = this.properties.getPrefix();
-		if (path.startsWith(prefix) && this.properties.isStripPrefix()) {
+		if (path.startsWith(prefix + "/") && this.properties.isStripPrefix()) {
```

Comment:
What if `prefix` already ends with a `/`?

Reply:
If prefix ended with `/`, than we would creating a path that looked like `/test//testservicea/**`.  Seems like that is invalid to begin with.

Reply:
ok. Do we prevent that from happening somewhere? If not, we should log an issue.

Reply:
Otherwise, LGTM

Reply:
done #2625